language-plutus-core/generators/Language/PlutusCore/Generators/Internal/TypeEvalCheck.hs

```diff
@@ -85,18 +94,19 @@ typeEvalCheckBy eval (TermOf term x) = TermOf term <$> do
     let valExpected = case makeKnown x of
             Error _ _ -> EvaluationFailure
             t         -> EvaluationSuccess t
-    fmap (TypeEvalCheckResult termTy) $ do
-        let valActual = eval term
-        if valExpected == valActual
-            then return valActual
-            else throwError $ TypeEvalCheckErrorIllEvaled valExpected valActual
+    fmap (TypeEvalCheckResult termTy) $ case eval term of
+        Right valActual
+            | valExpected == valActual -> return valActual
+            | otherwise                ->
+                throwError $ TypeEvalCheckErrorIllEvaled valExpected valActual
+        Left exc -> throwError $ TypeEvalCheckErrorException $ show exc
 
 -- | Type check and evaluate a term and check that the expected result is equal to the actual one.
 -- Throw an error in case something goes wrong.
 unsafeTypeEvalCheck
     :: forall a. KnownType a => TermOf a -> TermOf EvaluationResultDef
 unsafeTypeEvalCheck termOfTbv = do
-    let errOrRes = typeEvalCheckBy evaluateCk termOfTbv
+    let errOrRes = typeEvalCheckBy (pureTry @CkMachineException . evaluateCk) termOfTbv
```

Comment:
Right, I was going to ask why we don't just use the non-throwing version, but there isn't one for the CK machine.

Reply:
We could just write `pure . evaluateCk` (and specify somewhere that the possible exception is `Void`, i.e. we do not really catch anything), because `hedgehog` for some reason doesn't swallow exceptions when they're thrown by the CK machine. But I thought I just do it in a slightly more sensible way.